dvc/repo/add.py

```diff
@@ -17,49 +27,21 @@ def add(repo, fname, recursive=False, no_commit=False):
     return stages
 
 
-def _collect_filenames_to_add(repo, fname, recursive):
-    if recursive and os.path.isdir(fname):
-        fnames = _collect_valid_filenames_from_directory(repo, fname)
-    else:
-        fnames = [fname]
-    return fnames
-
-
-def _collect_valid_filenames_from_directory(repo, fname):
-    fnames = []
-    for file_path in _file_paths(fname):
-        if _is_valid_file_to_add(file_path, repo):
-            fnames.append(file_path)
-    return fnames
-
-
-def _file_paths(directory):
-    for root, _, files in os.walk(directory):
-        for f in files:
-            yield os.path.join(root, f)
-
-
-def _is_valid_file_to_add(file_path, repo):
-    if Stage.is_stage_file(file_path):
-        return False
-    if os.path.basename(file_path) == repo.scm.ignore_file:
-        return False
-    if repo.scm.is_tracked(file_path):
-        return False
-    return True
-
-
-def _create_stagefiles(repo, fnames, no_commit):
+def _create_stages(repo, targets, fname, no_commit):
     stages = []
+
     with repo.state:
-        for f in fnames:
-            stage = Stage.create(repo=repo, outs=[f], add=True)
+        for out in targets:
+            stage = Stage.create(repo=repo, outs=[out], add=True, fname=fname)
```

Comment:
This is wrong. It overwrites dvc file `fname` for each target. We should forbid `-R` and `-f` combination, same as we forbid multiple targets.

Reply:
@mroutis What about this one though?

Reply:
I put a restriction right here: https://github.com/iterative/dvc/pull/1671/files#diff-3e8e83df18c492b2100df4c29bd94a1dR13

Reply:
There's also a test: https://github.com/iterative/dvc/pull/1671/files#diff-db60e57582e26c8e55b0bc7f331089e9R412

Reply:
@mroutis Sorry, didn't notice that right away :slightly_smiling_face: